jobsets/default-mantis.nix

```diff
@@ -1,47 +1,141 @@
+            branch = declInput.rev;
+          };
+
+          nixpkgs = mkGitSrc {
+            repo = "https://github.com/NixOS/nixpkgs-channels";
+            branch = "refs/heads/nixos-18.03";
+          };
+        } // extraInputs;
+
+        enabled = 1;
+        hidden = false;
+        checkinterval = 90;
+        schedulingshares = 100;
+        emailoverride = "";
+        enableemail = false;
+        keepnr = 3;
+      };
     };
-  };
-  mainJobsets = with pkgs.lib; mapAttrs (name: settings: defaultSettings // settings) (rec {
-    mantis-testnet = mkMantis "phase/iele_testnet";
-  });
-  jobsetsAttrs = mainJobsets;
-  jobsetJson = pkgs.writeText "spec.json" (builtins.toJSON jobsetsAttrs);
+
+    mkMantisJob = {
+      name,
+      description,
+      mantisBranch ? "refs/heads/phase/iele_testnet",
+      solidityServiceBranch ? "refs/heads/master"
+    }:
+      mkJob {
+        inherit name description;
+        nixexprpath = "jobsets/release-mantis.nix";
+        extraInputs = {
+          sbtVerifySrc = mkGitSrc {
+            repo = "https://github.com/input-output-hk/sbt-verify.git";
+            branch = "refs/tags/v0.4.1";
+          };
+          mantisSrc = mkGitSrc {
+            repo = "https://github.com/input-output-hk/mantis.git";
+            branch = mantisBranch;
+            deepClone = true;
+          };
+          kevmSrc = mkGitSrc {
+            repo = "https://github.com/kframework/evm-semantics.git";
+            branch = "cc9c659510383215e5333d7f60088a2fdf2d120d";
+          };
+          secp256k1Src = mkGitSrc {
+            repo = "https://github.com/bitcoin-core/secp256k1";
+          };
+          ieleSrc = mkGitSrc {
+            repo = "https://github.com/runtimeverification/iele-semantics.git";
+            branch = "f494469ddeceb356dc9ef2b654e136c38c14722b";
+          };
+          soliditySrc = mkGitSrc {
+            repo = "https://github.com/runtimeverification/solidity.git";
+            branch = "refs/heads/sol2iele";
+          };
+          solidityServiceSrc = mkGitSrc {
+            repo = "https://github.com/input-output-hk/solidity-service.git";
+            branch = solidityServiceBranch;
+            deepClone = true;
+          };
+          solcBinSrc = mkGitSrc {
+            repo = "https://github.com/ethereum/solc-bin.git";
+            branch = "refs/heads/gh-pages";
+          };
+          remixIdeSrc = mkGitSrc {
+            repo = "https://github.com/input-output-hk/remix-ide.git";
+            branch = "refs/heads/phase/iele_testnet";
+          };
+        };
+      };
+
+    mantisJobsetDefinition = pkgs.lib.listToAttrs (
+      [
+        (mkMantisJob {
+          name = "iele_testnet";
+          description = "IELE Testnet";
+        })
+      ]
+      ++
+      (pkgs.lib.mapAttrsToList
+        (
+          num:
+          info: mkMantisJob {
+            name = "mantis-PR-${num}";
+            description = info.title;
+            mantisBranch = info.head.sha;
```

Comment:
this means that a PR jobset cant auto-detect new pushes to the PR, it should be changed to use `pull/${num}/head` like cardano is using